### PR TITLE
Secret sdk internal

### PR DESCRIPTION
**Pull request by ram-infrac**
**What this PR does / why we need it**:
Internal review 
**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: api/client/cluster/client.go [hunk 1]
```diff
@@ -192,3 +193,68 @@ func (c *clusterClient) EraseAlert(resource api.ResourceType, alertID int64) err
 	}
 	return nil
 }
+
+func (c *clusterClient) SetClusterSecretKey(secretKey string, override bool) error {
+	request := c.c.Post().Resource("/setclustersecretkey")
+	request.QueryOption(secrets.ClusterSecretKey, secretKey)
```
Comment (ipochi) [inline api/client/cluster/client.go:199]: is that double c a typo [c.c] , if not then give proper name , its confusing.

if the chain of calls is long make it like this

https://github.com/libopenstorage/openstorage/blob/6a6f2b47c4940f90435ab80d810a566208326e43/api/server/cluster_test.go#L93

File: secrets/struct.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+// swagger: parameters clusterKey
+type ClusterSecretKeyRequest struct {
+	Clustersecretkey string
+	Override         bool
+}
+
+// SetsecretsLogin setsecrets
+// swagger: parameters secret
+type SetSecretsRequest struct {
+	Secretid    string
```
Comment (ipochi) [inline secrets/struct.go]: SecretID
make sure all variables are camelCase or CamelCase in case of export

File: secrets/secrets.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusNotImplemented)
+		//c.sendError(c.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.Write([]byte(secretKeyOkMsg + "\n"))
+}
+
+//TODO: Add swagger yaml
```
Comment (ipochi) [inline secrets/secrets.go:140]: you can submit the PR without swagger.
later on you can but make sure you do.

File: secrets/fake/fake.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+func (f *Fake) CheckSecretLogin() error {
+	return ErrNotImplemented
+}
+
+func (f *Fake) SetSecret(secretKey string, secretValue string) error {
+	return ErrNotImplemented
+}
+
+func (f *Fake) GetSecret(secretKey string) (string, error) {
+	return "", ErrNotImplemented
```
Comment (ipochi) [inline secrets/fake/fake.go:40]: this has been confirmed with luis  ? 
regarding returning empty string or nil

File: api/server/server.go [hunk 5]
```diff
@@ -157,6 +165,7 @@ func startServer(name string, sockBase string, port uint16, routes []*Route) err
 
 type restServer interface {
 	Routes() []*Route
+	RegisterSecretManager(secret secrets.SecretManager) *secrets.Manager
```
Comment (ipochi) [inline api/server/server.go:168]: lets have a discussion on this luis after PR
this doesnt look like a place for RegisterSecretManager

File: api/server/cluster.go [hunk 6]
```diff
@@ -711,3 +712,8 @@ func handleResourceType(resource string) (api.ResourceType, error) {
 		return api.ResourceType_RESOURCE_TYPE_NONE, fmt.Errorf("Invalid resource type")
 	}
 }
+
+func (c *clusterApi) RegisterSecretManager(sec secrets.SecretManager) *secrets.Manager {
```
Comment (ipochi) [inline api/server/cluster.go:716]: this might need a different location

File: api/client/cluster/client.go [hunk 7]
```diff
@@ -192,3 +193,68 @@ func (c *clusterClient) EraseAlert(resource api.ResourceType, alertID int64) err
...
+	resp := request.Do()
+	if resp.Error() != nil {
+		return resp.FormatError()
+	}
+	return nil
+}
+
+func (c *clusterClient) SetSecret(secretid string, secretvalue string) error {
+	request := c.c.Post().Resource("/setsecret")
+	request.QueryOption(secrets.SecretKey, secretid)
```
Comment (ipochi) [inline api/client/cluster/client.go]: create a secretrequest and secret response struct
marshal the data in the struct instance
and send the instance in the Body

Comment (ram-infrac) [reply api/client/cluster/client.go]: I did but internal PX secret implemenation is such way that it accept  secretid as params and data as body.But will ask aditya about same

File: api/server/cluster.go (hunk 6 above)
Comment (ram-infrac) [reply api/server/cluster.go:716]: Have to discuss this with luis

File: api/server/server.go (hunk 5 above)
Comment (ram-infrac) [reply api/server/server.go:168]: I agree

File: api/client/cluster/client.go (hunk 1 above)
Comment (ram-infrac) [reply api/client/cluster/client.go:199]: I kept it uniform with other post()/get() call api/cluster/client.go

File: secrets/struct.go (hunk 2 above)
Comment (ram-infrac) [reply secrets/struct.go]: ok

File: secrets/secrets.go (hunk 3 above)
Comment (ram-infrac) [reply secrets/secrets.go:140]: ok Imran. I already did for secretLogin() api, once this PR is reviewed by luis and aditya I will add for every other API

File: secrets/fake/fake.go (hunk 4 above)
Comment (ram-infrac) [reply secrets/fake/fake.go:40]: What do you think we should return here ?

**Pull request merged**